starknet_devnet/predeployed_contract_wrapper.py

```diff
@@ -0,0 +1,9 @@
+"""Parent for predeployed contract wrapper classes"""
+
+
+class PredeployedContractWrapper:
+    """Parent class for all predeployed contract wrapper classes"""
+
+    async def deploy(self):
```

Comment:
I would expect this base class to also do some common logic, otherwise I don't see a lot of use from this abstraction. I.e. this could be a template method (which does e.g. starknet state manipulation) and there could be some unimplemented hook methods (e.g. one could do constructor mocking).

Reply:
What do you think of something like this:
```python
class PredeployedContractWrapper:
    """Parent class for all predeployed contract wrapper classes"""

    async def deploy_contract(self):
        # pylint: disable=protected-access
        starknet: Starknet = self.starknet_wrapper.starknet
        starknet.state.state.cache._class_hash_writes[
            self.address
        ] = self.class_hash_bytes
        # replace with await starknet.state.state.deploy_contract

    async def mimic_constructor(self):
        raise NotImplementedError()

    async def deploy(self):
        """Deploy the contract wrapper to devnet"""
        starknet: Starknet = self.starknet_wrapper.starknet

        await starknet.state.state.set_contract_class(
            self.class_hash_bytes, self.contract_class
        )

        await self.deploy_contract()

        await self.mimic_constructor()

        self.contract = StarknetContract(
            state=starknet.state,
            abi=self.contract_class.abi,
            contract_address=self.address,
            deploy_call_info=None,
        )
```